### PR TITLE
Rt 844 forget password

### PR DESCRIPTION
**Pull request by iandjx**
# Description

Please include a summary of the changes and the related issue in checklist format. Please also include relevant motivation and context.

- [x] Added reset password functionality
- [x] switch to firebase admin sdk
- [x] removed app context
- [x] store account reference as http only cookie and used serverside to make api calls to banxa
- [x] use nodemailer to send password reset link

## JIRA Issue
https://romeblockchain.atlassian.net/browse/RT-844?atlOrigin=eyJpIjoiYTZjZTk3MGY5NmFkNGYxMmIwMWNiNzIyZTI5OGRiZDAiLCJwIjoiaiJ9

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
https://banxa-sandbox-h5axcfkrgq-uc.a.run.app/

- [x] register new account
- [x] logout
- [x] login using credentials
- [x] logout
- [x] click forgot password
- [x] enter email
- [x] should receive email
- [x] should be redirected to reset password page
- [x] enter new password

# Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: banxa/Context/AuthContext.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React, { createContext, ReactNode, useEffect } from 'react'
+import useLocalStorage from 'utils/useLocalStorage'
+import { useRouter } from 'next/router'
+
+interface IAuthContext {
+  isLoggedIn: string
+  setIsLoggedIn: (val: string) => void
+}
+export const AuthContext = createContext<IAuthContext>({
+  isLoggedIn: 'false',
```
Comment (MarkCBall) [inline banxa/Context/AuthContext.tsx:10]: could this be a boolean or an enum instead of a string?

Comment (iandjx) [reply banxa/Context/AuthContext.tsx:10]: Local storage can only accept strings.

https://stackoverflow.com/questions/3263161/cannot-set-boolean-values-in-localstorage

 Enum can be a good alternative

File: banxa/pages/api/reset-password.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  if (expired) {
+    await document.ref.update({
+      resetToken: null,
+      expiryDate: null,
+    })
+
+    return res.status(400).send('Token has expired')
+  }
+
+  const salt = await bcrypt.genSalt(10)
```
[the remark this answers is not in the document]
Comment (iandjx) [reply banxa/pages/api/reset-password.ts:28]: https://stackoverflow.com/questions/6832445/how-can-bcrypt-have-built-in-salts

[the remark this answers is not in the document]
Comment (MarkCBall) [reply banxa/pages/api/reset-password.ts:28]: Do we have any documentation saved on confluence for how banxa works?

File: banxa/pages/privacy-policy.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
+import { ChevronLeftIcon } from '@heroicons/react/solid'
+import { useRouter } from 'next/router'
+import React from 'react'
+
+const PrivacyPolicy = () => {
```
Comment (MarkCBall) [inline banxa/pages/privacy-policy.tsx:5]: this is great! Where is this copy from? should anyone review it ect?

**Review by MarkCBall — APPROVED** [review]

File: banxa/pages/api/reset-password.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (iandjx) [reply banxa/pages/api/reset-password.ts:28]: > is this documented somewhere? Confluence ect?

Ill add it in confluence :)

File: banxa/Context/AuthContext.tsx (hunk 1 above)
Comment (iandjx) [reply banxa/Context/AuthContext.tsx]: switched to enum AUTH_STATUS

File: banxa/components/auth/Login.tsx [hunk 4]
```diff
@@ -105,12 +103,15 @@ const Login = ({
...
-
-      <button onClick={() => setIsLogin(false)} className="mt-4 text-base">
-        No Account? Sign up here
+      <button
+        onClick={() => router.push('/forgot-password')}
+        className="mt-2 text-sm hover:underline  text-left underline-offset-4"
+      >
+        Forgot your password?
+      </button>
+      <button onClick={() => setIsLogin(false)} className="mt-4 text-base hover:underline  underline-offset-4">
```
[the remark this answers is not in the document]
Comment (iandjx) [reply banxa/components/auth/Login.tsx]: change setislogin to setisloginpage

**Pull request merged**